scapy/layers/snmp.py

```diff
@@ -248,12 +248,12 @@ def snmpwalk(dst, oid="1", community="public"):
     try:
         while True:
             r = sr1(IP(dst=dst)/UDP(sport=RandShort())/SNMP(community=community, PDU=SNMPnext(varbindlist=[SNMPvarbind(oid=oid)])),timeout=2, chainCC=1, verbose=0, retry=2)
-            if ICMP in r:
-                print(repr(r))
-                break
             if r is None:
                 print("No answers")
                 break
+            if ICMP in r:
+                print(repr(r))
```

Comment:
How about ``

Reply:
Hm.. My "knowledge" of net is too narrow :(.
It is only simple fix to avoid:
    if ICMP in r:
TypeError: argument of type 'NoneType' is not iterable
where is no answer.
I will be grateful for any suggestions.

Reply:
Also help me please to chose appropriate destination ip for test on Travis for:
snmpwalk(dst="0.0.0.0")  # Crashed on commit: c2acc60e985340884b50bcf05764dcb4eabea1dc
in case when it works in "no sudo" mode e.g.:
https://travis-ci.org/secdev/scapy/jobs/351828731
or how to avoid this test for this case?